Add poco.

sdcb opened the pull request:
This should fix: https://github.com/Microsoft/vcpkg/issues/124

Note:
I created a small patch to define POCO_NO_AUTOMATIC_LIBS since vcpkg would do that.
The option POCO_UNBUNDLED is now OFF, we should switch it ON when all dependency is available in vcpkg.

sdcb committed:
Add poco.

msftclas commented:
Hi __@sdcb__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

sdcb committed:
[poco] Remove useless depends, Add 7zip, drop POCO_MT.

sdcb commented:
Explain:
1. I removed the depends on "openssl" and "libmariadb" since it does not work as I expected
2. I switched to use the github.com release instead of the pocoproject.org windows release because that can enable more option
3. I switched the ENABLE_SEVENZIP on, I think it may be useful for someone.

Note:
1. I'm not very similar for building static libraries, I did not switch the POCO_MT on because I don't know what that means. Please feel free to do any change when merging
2. The lib Poco/Data/MySql won't build even if add libmariadb dependency.

ras0219-msft commented:
I think we have all the dependencies, so I'm going to try switching UNBUNDLED to ON and merge that if possible. The libraries that appear to be affected by that switch are:
- zlib
- pcre
- expat
- sqlite3

Additionally, it looks like it can optionally use:
- openssl (`openssl` in vcpkg)
- apr (`apr` in vcpkg)
- apache2
- mysql (`libmysql` or `libmariadb` in vcpkg)
- odbc

ras0219-msft commented:
Thanks for the PR! I made the change to use unbundled libraries.

I needed to also modify PCRE to pre-pre-process the value of PCRE_STATIC into the headers, so the `PCRE_STATIC` macro now does nothing and the correct PCRE library is loaded based on how PCRE was built. Make sure to update it if you have it already built!
